src/graphics/prefilter-cubemap.js

```diff
@@ -41,9 +41,9 @@ Object.assign(pc, (function () {
         var format = sourceCubemap.format;
 
         var cmapsList = [[], options.filteredFixed, options.filteredRgbm, options.filteredFixedRgbm];
-        var gloss = method === 0 ? [0.9, 0.85, 0.7, 0.4, 0.25] : [512, 128, 32, 8, 2]; // TODO: calc more correct values depending on mip
-        var mipSize = [64, 32, 16, 8, 4]; // TODO: make non-static?
-        var numMips = 5;
+        var gloss = method === 0 ? [0.9, 0.85, 0.7, 0.4, 0.25, 0.15, 0.1] : [512, 128, 32, 8, 2, 1, 1]; // TODO: calc more correct values depending on mip
+        var mipSize = [64, 32, 16, 8, 4, 2, 1]; // TODO: make non-static?
+        var numMips = 7;                        // generate all mips down to 1x1
```

Comment:
Do changes to this file mean that generated prefiltered cubemaps now have 7 mips instead of 5? And all are now written to DDS when prefiltering in the Editor? Just thinking about all the project out there with cubemap DDSs with the old incomplete set of mips. It looks like your code handles both cases but just wanted to check.

Reply:
Yup that's all correct.